Whiskey/Tasks/Set-WhiskeyProGetAsset.ps1

```diff
@@ -0,0 +1,97 @@
+        * `Name` (Mandatory): The desired Name you wish the file to be named in ProGet.
+        * `Directory` (Mandatory): The Path to the Directory you wish to upload the asset to.
+        * `ApiKey` (Mandatory): The ApiKey to Proget Api.
+        * `ProGetUsername` (Mandatory): The Username to the ProGet Api.
+        * `ProGetPassword` (Mandatory): The Password to the ProGet Api.
+
+        .EXAMPLE       
+        BuildTasks:
+        - SetProGetAsset:
+            Path: 'path/to/file.txt'
+            Uri: http://proget.dev.webmd.com/
+            Name: 'exampleAsset'
+            Directory: 'versions'
+            ApiKey: 'exampleApiKey'
+            ProGetUsername: Admin
+            ProGetPassword: Admin
+
+        Example of adding an asset to ProGet in the versions directory.      
+    #>
+    [Whiskey.Task("SetProGetAsset",SupportsClean=$true)]
+    [CmdletBinding()]
+    param(
+        [object]
+        # The context this task is operating in. Use `New-WhiskeyContext` to create context objects.
+        $TaskContext,
+        
+        [hashtable]
+        # The parameters/configuration to use to run the task.
+        $TaskParameter
+    )
+
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    if( -not $TaskParameter['Name'] )
+    {
+        Stop-WhiskeyTask -TaskContext $TaskContext -Message ("Please add a valid Name to your whiskey.yml file: 
+        BuildTasks:
+        - SetProGetAsset:
+            Path: 'path/to/file.txt'
+            Uri: http://proget.dev.webmd.com/
+            Name: 'exampleAsset'
+            Directory: 'versions'
+            ApiKey: 'exampleApiKey'
+            ProGetUsername: Admin
+            ProGetPassword: Admin
+            ")
+    }
+
+    if( -not $TaskParameter['Directory'] )
+    {
+        Stop-WhiskeyTask -TaskContext $TaskContext -Message ("Please add a valid Directory to your whiskey.yml file'
+        BuildTasks:
+        - SetProGetAsset:
+            Path: 'path/to/file.txt'
+            Uri: http://proget.dev.webmd.com/
+            Name: 'exampleAsset'
+            Directory: 'versions'
+            ApiKey: 'exampleApiKey'
+            ProGetUsername: Admin
+            ProGetPassword: Admin")
+    }
+
+    if( -not $TaskParameter['ApiKey'] -or  -not $TaskParameter['ProGetUsername'] -or -not $TaskParameter['ProGetPassword'])
+    {
+        Stop-WhiskeyTask -TaskContext $TaskContext -Message ("Please add valid ProGet credentials and ApiKey to your whiskey.yml file
+        BuildTasks:
+        - SetProGetAsset:
+            Path: 'path/to/file.txt'
+            Uri: http://proget.dev.webmd.com/
+            Name: 'exampleAsset'
+            Directory: 'versions'
+            ApiKey: 'exampleApiKey'
+            ProGetUsername: Admin
+            ProGetPassword: Admin")
+    }
+
+    $credential = New-Credential -UserName $TaskParameter['ProGetUsername'] -Password $TaskParameter['ProGetPassword']

```

Comment:
You would use the `Get-WhiskeyCredential` or `Get-WhiskeyApiKey` functions to get the credential and/or API key. Does ProGet require both? Or can it use one or the other?

Reply:
In order to check if the asset directory exists using the `Test-ProGetFeed` function we need the api key, in order to add the asset we need the credentials. So we need both unless we want to remove the write-error displaying the asset directory doesn't exist.

The error that is responded without the directory check is `The remote server returned an error: (404) Not Found.`

Reply:
Done